src/test/streams_tests.cpp

```diff
@@ -329,12 +329,80 @@ BOOST_AUTO_TEST_CASE(streams_buffered_file)
     // by the rewind window (relative to our farthest read position, 40).
     BOOST_CHECK(bf.GetPos() <= 30);
 
+    bf.FindByte(31);
+    BOOST_CHECK_EQUAL(bf.GetPos(), 31U);
+
+    // If the byte isn't found, end of file.
+    try {
+        bf.FindByte(99);
+        BOOST_CHECK(false);
+    } catch (const std::exception& e) {
+        BOOST_CHECK(strstr(e.what(),
+                           "CBufferedFile::Fill: end of file") != nullptr);
+    }
+    BOOST_CHECK_EQUAL(bf.GetPos(), 40U);
+    BOOST_CHECK(bf.eof());
+
+    // Test Search(), it leaves the postion just after the found bytes.
+    BOOST_CHECK(bf.SetPos(30));
+    {
+        unsigned char needle[2] = {32, 33};
+        bf.Search(needle, 2);
+    }
+    BOOST_CHECK_EQUAL(bf.GetPos(), 34U);
+
+    // The bytes must be consecutive, this won't be found.
+    {
+        unsigned char needle[2] = {36, 38};
+        try {
+            bf.Search(needle, 2);
+            BOOST_CHECK(false);
+        } catch (const std::exception& e) {
+            BOOST_CHECK(strstr(e.what(),
+                               "CBufferedFile::Fill: end of file") != nullptr);
+        }
```

Comment:
134de90f2b64ede6dbd812a9a88b00708685394d
```suggestion
        BOOST_CHECK_EXCEPTION(bf.Search(needle, 2), std::ios_base::failure, HasReason("CBufferedFile::Fill: end of file"));
```